bin/improver-spot-extract

```diff
@@ -122,6 +149,40 @@ def main():
         grid_metadata_identifier=args.grid_metadata_identifier)
     result = plugin.process(neighbour_cube, diagnostic_cube)
 
+    # If a probability or percentile diagnostic cube is provided, extract
+    # the given percentile if available. This is done after the spot-extraction
+    # to minimise processing time; usually there are far fewer spot sites than
+    # grid points.
+    if args.extract_percentile:
+        try:
+            perc_coordinate = find_percentile_coordinate(result)
+        except CoordinateNotFoundError:
+            if 'probability_of_' in result.name():
+                result = GeneratePercentilesFromProbabilities(
+                    ecc_bounds_warning=args.ecc_bounds_warning).process(
+                        result, percentiles=args.extract_percentile)
+                result = iris.util.squeeze(result)
+            else:
+                msg = ('Diagnostic cube is not a known probabilistic type. '
+                       'The {} percentile could not be extracted. Extracting '
+                       'data from the cube including any leading '
+                       'dimensions.'.format(
+                           args.extract_percentile))
+                if not args.suppress_warnings:
+                    warnings.warn(msg)
+        else:
+            constraint = ['{}={}'.format(perc_coordinate.name(),
```

Comment:
What if we need to resample percentiles?

Reply:
We cannot do this currently with percentile-extract or extract, which are the two ways in which spot data is currently obtained. We don't have a CLI to do this explicitly either, the percentiles-to-realizations CLI effectively does it, but rebadges the results as realizations. As such, this would be functionality we don't currently have or use, but which I can see use cases for.

Do you want me to add it? It will make use of private functions within the ResamplePercentiles plugin in the easiest implementation, for example:
```
from improver.ensemble_copula_coupling.ensemble_copula_coupling_utilities import get_bounds_of_distribution
from improver.ensemble_copula_coupling.ensemble_copula_coupling import ResamplePercentiles
bounds = get_bounds_of_distribution(cube.name(), cube.units)
result = ResamplePercentiles()._interpolate_percentiles(cube, desired_percentiles, bounds, 'percentile_over_realization')
```
Alternatively we could modify the ResamplePercentiles plugin to accept a list of percentiles rather than just a number and use that directly.

Both implementations are a bit more work than I was striving for here. Perhaps we can spin this off into a ticket for additional functionality that we don't yet need/use?

Reply:
What we are really doing with spot extraction at the moment is extracting a psuedo-deterministic product by extracting the 50% percentile. We could think of this in two ways:
* At the moment we rely on the 50% percentile being present in the input in order to extract it. 
* In the longer term will we need to be able to extract other things at the spot extraction step? Things like specific thresholds or percentile for products. I don't know if we need to do this or not. It would be quite easy to do but is out of scope at the moment. 
If we need to do both in the longer term, then it seems a good idea to do some work soon to modify the ResamplePercentiles plugin to take in specific percentiles and it might be worth doing anyway to avoid the first problem. But seeing as we don't quite have the functionality yet I think it probably is out of scope and should be another ticket if we need it. We will need to update the gridded processing to be consistent as well if we choose to do it.

Reply:
Post-poned for now. We may need to revisit if the need arises.

Reply:
I'm happy to leave this for now.